xacro/__init__.py

```diff
@@ -171,41 +181,58 @@ def check_deprecated_tag(tag_name):
 
 
 class Macro(object):
+
     def __init__(self):
         self.body = None  # original xml.dom.Node
         self.params = []  # parsed parameter names
         self.defaultmap = {}  # default parameter values
         self.history = []  # definition history
 
 
+class ResourceNotFound(Exception):
```

Comment:
I wouldn't define this here; it's too easy for it to get forgotten about later.  Instead I'd rather remove or comment out the `ResourceNotFound` section below.